[IMP] website: allow cross-company records access to portal user

tegu-odoo opened the pull request:
### Purpose:
To allow end-customers(portal users) to check documents from multiple companies using single portal account.

### Before this PR:
Even when the admin activates the "Shared Customer Account" option from the settings of the website, records (orders, tickets, etc.) related only to the default company of the user remain visible on website.

### After this PR:
Now, when admin activates 'Shared Customer Account' option then portal user will be able to access records (orders, tickets, etc.) of all allowed companies from single website.

robodoo commented:
[Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/167042).

thsh-odoo commented:
Hello @tegu-odoo 
A small unit test would be nice :smiley: 
Thanks!!

tegu-odoo commented:
> Hello @tegu-odoo A small unit test would be nice 😃 Thanks!!

Hello @thsh-odoo 
For this task I have fixed in _eval_context function defined in inherited ir.rule model in website module. In this function i have modified value for company_ids. As it is not updating anything in database it only updating return value of _eval_context function. So, it is not possible write testcase for this fix in website module.  Also it is not possible to write testcase to compare response of _eval_context. But we can write testcase for it in another modules which are company dependents like sale, leads, etc.,  Also as per our internal discussion changes are done.
Thanks!!
